pkgs/by-name/ht/htb-toolkit/package.nix

```diff
@@ -0,0 +1,85 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, openssl
+, stdenv
+, darwin
+, coreutils
+, noto-fonts-color-emoji
+, gnome
+, libsecret
+, bash
+, openvpn
+, nerdfonts
+, gzip
+, killall
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "htb-toolkit";
+  version = "unstable-2024-01-17";
+
+  src = fetchFromGitHub {
+    owner = "D3vil0p3r";
+    repo = "htb-toolkit";
+    # https://github.com/D3vil0p3r/htb-toolkit/issues/3
+    rev = "54e11774ea8746ea540548082d3b25c22306b4fc";
+    hash = "sha256-QYUqdqFV9Qn+VbJTnz5hx5I0XV1nrzCoCKtRS7jBLsE=";
+  };
+
+  cargoHash = "sha256-XDE6A6EIAUbuzt8Zb/ROfDAPp0ZyN0WQ4D1gWHjRVhg=";
+
+  # Patches only relevant for Nixpkgs
+  patches = [
+    ./main.patch
+    ./manage.patch
+    ./play.patch
+    ./types.patch
+    ./utils.patch
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    openssl
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
+  ];
+
+  propagateBuildInputs = [
+    (nerdfonts.override { fonts = [ "NerdFontsSymbolsOnly" ]; })
+    coreutils
+    noto-fonts-color-emoji
```

Comment:
I fail to see how this package depends on these fonts. Is it because it uses weird unicode characters? I'm not sure they will be reachable by the font system for other packages (like a terminal emulator) if you include it like this.

Reply:
It should use these packages because when you use the tool and it prints the table with the list of free/starting point/retired machines, the glyphs related to Windows and Linux machines come from `NerdFontsSymbolsOnly`. Probably I can remove `noto-fonts-color-emoji`.

Reply:
That makes sense. But I still don't think this would work. There would be left a reference to the path to the font in `$out/nix-support`, but the terminal emulator of the user wouldn't be able to pick it up by itself.

I don't really know of any good solution for this. Maybe a nixos `programs.htb-toolkit` module, ensuring that the font is included in the system fonts or something?

Reply:
Yes. I think you are right. For now I remove that, because actually it has no effect. Thanks